app/Http/Controllers/Api/ApplicantController.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\UpdateApplicantProfile;
+use App\Http\Resources\ApplicantProfile as ApplicantProfileResource;
+use App\Models\Applicant;
+use App\Models\ApplicantClassification;
+
+class ApplicantController extends Controller
+{
+
+    /**
+     * Retrieve Applicant profile.
+     *
+     * @param Applicant $applicant Incoming Applicant object.
+     *
+     * @return mixed
+     */
+    public function getProfile(Applicant $applicant)
+    {
+          $applicant->loadMissing('applicant_classifications');
+          return new ApplicantProfileResource($applicant);
+    }
+
+    /**
+     * Update Applicant profile.
+     *
+     * @param UpdateApplicantProfile $request Form Validation casted request object.
+     * @param Applicant              $applicant Incoming Applicant object.
+     */
+    public function updateProfile(UpdateApplicantProfile $request, Applicant $applicant)
+    {
+        $validatedRequest = $request->validated();
+        // If there are no applicant classifications in the request,
+        // then delete all applicant classifications attached to applicant.
+        if (!array_key_exists('applicant_classifications', $validatedRequest)) {
+            $applicant->applicant_classifications()->delete();
+        } else {
+            $newApplicantClassifications = collect($validatedRequest['applicant_classifications']);
+            $oldApplicantClassifications = $applicant->applicant_classifications;
+
+            // Delete old applicant classifications that were not resubmitted.
+            foreach ($oldApplicantClassifications as $applicantClassification) {
+                $newApplicantClassification = $newApplicantClassifications->firstWhere(
+                    'id',
+                    $applicantClassification['id']
+                );
+                if (!$newApplicantClassification) {
+                    $applicantClassification->delete();
+                }
+            }
+
+
+            // Update old applicant classifications and/or create them if it doesn't exist.
+            // This should remove any duplicate classification-level combinations as well.
+            $newApplicantClassifications->map(function ($applicantClassification) {
+                $newApplicantClassification = ApplicantClassification::firstOrNew(
+                    [
+                        'classification_id' => $applicantClassification['classification_id'],
+                        'level' => $applicantClassification['level']
+                    ]
+                );
```

Comment:
Checking against `id` when deleting, and against `classification_id` and `level` when creating/updating can create some discrepancies. I wasn't a hundred percent sure about this, but I added a test (see 747d8b5863ad67d64273857ac45df3d023f91701) to confirm.

If you re-submit an existing application_classification (including with existing id) but with classification_id or level changed, then this _won't_ delete the previous version (correct) but it will create a _new_ object with `firstOrNew` resulting in the old and new versions both existing in the DB.

While `firstOrNew` seems convenient here, I think it would be more reliable to just check if the id exists, manually create a new object if it doesn't, and then update the values either way.

Reply:
Just pushed up a fix now